### PR TITLE
feat: btrfs based storage quotas

### PR DESCRIPTION
**Pull request by noxiouz**
### How it works

It's based on **Btrfs** subvolumes and quota groups limits. Each subvolume can have its own limit (Docker sets this limit if you ask) and several subvolumes can be added to a quota group to have shared limit. https://btrfs.wiki.kernel.org/index.php/Quota_support Playing with quota groups allows us to provide a shared limit. 

When a container is created we lookup its rootfs directory. *WARNING: Docker does not expose this API, so some dark magic applied to fetch btrfs graph driver related information.*
We extract this information by reading files inside graph driver directory.

DealID is used as quota group id. We allocate new group if it does not exist. A subvolume with roofs of the container is assigned to this group. When container dies we remove this subvolume from quota group. If the quota group is empty, it's deallocated. 

### Concept

https://gist.github.com/noxiouz/3e4fc753862aa8d157b8f764ad2b025e

### CODE REVIEW

**Comment by noxiouz** [comment]
Yes, I will add it.

File: insonmnia/worker/storagequota/btrfs/btrfs_api.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	// TODO: add implementation
+	output, err := exec.CommandContext(ctx, "btrfs", "qgroup", "show", path).Output()
+	if err != nil {
+		log.G(ctx).Error("failet to lookup quota", zap.String("path", path), zap.String("quota", qgroupID), zap.Error(err))
+	}
+
+	return lookupQuotaInShowOutpout(output, qgroupID)
+}
+
+func lookupQuotaInShowOutpout(output []byte, qgroupID string) (bool, error) {
```
Comment (oopcode) [inline insonmnia/worker/storagequota/btrfs/btrfs_api.go]: s/Outpout/Output/g

File: insonmnia/worker/storagequota/btrfs/btrfs_api.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	qgroupIDBytes := []byte(qgroupID)
+	r := bytes.NewReader(output)
+	scanner := bufio.NewScanner(r)
+	foundHeader := false
+	endOfHeaderToken := []byte("--------") // next line after qgroupid
```
Comment (oopcode) [inline insonmnia/worker/storagequota/btrfs/btrfs_api.go]: shiiiet

Comment (noxiouz) [reply insonmnia/worker/storagequota/btrfs/btrfs_api.go]: Look at tests :) do not know how to detect it

File: insonmnia/worker/storagequota/common.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+func (e ErrDriverNotSupported) Error() string {
+	return fmt.Sprintf("Driver %s not supported", e.driver)
+}
+
+type QuotaDescription struct {
+	Bytes uint64
+}
+
+type Cleanup interface {
+	Close() error
```
[the remark this answers is not in the document]
Comment (noxiouz) [reply insonmnia/worker/storagequota/common.go]: Interface is the same as for other cleanupers

**Comment by noxiouz** [comment]
Adding C API based implementation. It should be more stable, as parsing CLI output is not reliable soultion

**Pull request merged**